app/services/request_mentoring_on_solution.rb

```diff
@@ -9,6 +9,26 @@ def call
       return if user_track.mentoring_allowance_used_up?
     end
 
+    post_system_messages!
```

Comment:
I think I'd prefer this to be `post_system_messages! if solution.approved_by_system?` as it's clearer when reading the call method as to why we're creating system messages (and remove the guard below). If you disagree, I don't mind you leaving it as it is though

Reply:
Cool. Sounds good.